feat(perfmon): Add native performance monitoring docs

relaxolotl opened the pull request:
A lot of this has been blatantly copied from https://github.com/getsentry/sentry-docs/pull/4610's changes, and updated appropriately for native. The contents of the prose-heavy docs in this significantly differ from the rust PR, however.

Credit for a lot of this goes to @loewenheim.

loewenheim committed:
WIP docs for native perfmon

loewenheim committed:
Add pages about concurrency and errors

relaxolotl committed:
additional changes for native perf docs

relaxolotl committed:
may be useful to actually mention how to turn it on

relaxolotl committed:
address comments that were already made on rust PR

relaxolotl committed:
odds and ends

relaxolotl commented on src/includes/performance/distributed-tracing/native.mdx:
````diff
@@ -0,0 +1,45 @@
+The function is to be used in `sentry_transaction_iter_headers` as a callback.
+The following example uses `sentry_value_t` as the aggregate object:
+```c
+static void
+copy_headers_to(const char *key, const char *value, void *userdata)
+{
+    sentry_value_t *headers
+        = (sentry_value_t *)userdata;
+    sentry_value_set_by_key(*headers, key, value);
+}
+
+int main(int argc, char **argv) {
+  ...
+	// transaction to continue off of
+	sentry_transaction_context_t *tx_ctx
+        = sentry_transaction_context_new("honk", NULL);
+    sentry_transaction_t *tx = sentry_transaction_start(tx_ctx);
+
+	sentry_value_t *headers = sentry_value_new_object();
+	sentry_transaction_iter_headers(tx, copy_headers_to, (void *)headers);
+}
````
@native, probably most importantly @Swatinem: i'll probably need a sanity check on the code and the example on this file.

relaxolotl committed:
minor phrasing changes

relaxolotl committed:
consistency

relaxolotl committed:
more correct jsx

relaxolotl committed:
formatting

relaxolotl committed:
i am an expert at jsx

relaxolotl committed:
hail mary

relaxolotl commented:
putting this in draft until i can figure out how to fix hyperlink's complaints about various native backend pages linking to nonexistent sampling pages

loewenheim commented:
For context, the links it complains about are
```
  platforms/native/guides/breakpad/configuration/sampling
  platforms/native/guides/crashpad/configuration/sampling
  platforms/native/guides/minidumps/configuration/sampling
  platforms/native/guides/ue4/configuration/sampling
  platforms/native/guides/wasm/configuration/sampling
```

loewenheim committed:
WIP trying something

loewenheim committed:
trying something part deux

loewenheim committed:
Revert "trying something part deux"

This reverts commit 1c0874299f4067982a77e97d5e7d1dc4e09d29d7.

loewenheim committed:
Revert "WIP trying something"

This reverts commit 096e476b0f6f77fa10aec9b4b61a877bae0a9708.

relaxolotl committed:
hail mary 2

relaxolotl committed:
hail mary 3

imatwawana reviewed (commented):
I know this is still in draft, but I wanted to make some suggestions for simplifying the language and avoiding passive voice. Otherwise though, this looks great!

relaxolotl committed:
apply feedback

relaxolotl committed:
fill in gaps according to perf doc instructions

relaxolotl committed:
explicitly exclude native backends

relaxolotl committed:
codeblock cleanup

relaxolotl committed:
give users a heads up about the sampling function instead of just omitting it

relaxolotl committed:
don't manually newline

relaxolotl committed:
better titles on warnings

relaxolotl committed:
be very clear that the flag is needed

relaxolotl committed:
rename

relaxolotl commented:
fixed the links

relaxolotl commented:
@imatwawana thanks for the early review; I've addressed the feedback you've left behind while fixing up this branch.

I've also added in additional text during the process of repairing and getting this branch to build, which I'd appreciate your feedback on as well. They're mixed into what you've already reviewed, so here's a list of the files that could use a second look:
- src/includes/performance/always-inherit-sampling-decision/native.mdx
- src/includes/performance/enable-tracing/native.mdx
- src/includes/performance/retrieve-transaction/native.mdx
- src/includes/performance/uniform-sample-rate/native.mdx (the last ~3 lines)

Other than the last item, everything listed above are new files with a very small amount of content (1-2 sentences).

loewenheim committed:
Revert "rename"

This reverts commit 384e5848a3d8fe2e41926875553abf2d4b9319e1.

loewenheim committed:
Merge branch 'master' into native-perf

imatwawana commented on src/includes/performance/always-inherit-sampling-decision/native.mdx:
```diff
@@ -0,0 +1,3 @@
+<Alert level="warning" title="Sampling Functions">
+The Native SDK currently does not support sampling functions (<PlatformIdentifier name="traces-sampler" />).
```
```suggestion
<Alert level="warning" title="Sampling Functions">

The Native SDK currently does not support sampling functions (<PlatformIdentifier name="traces-sampler" />).

```

imatwawana commented on src/includes/performance/uniform-sample-rate/native.mdx:
```diff
@@ -0,0 +1,7 @@
+
+<PlatformContent includePath="performance/traces-sample-rate" />
+
+<Alert level="warning" title="Sampling Functions">
+The Native SDK currently does not support setting a sampling function (<PlatformIdentifier name="traces-sampler" />).
```
```suggestion
<Alert level="warning" title="Sampling Functions">

The Native SDK currently does not support setting a sampling function (<PlatformIdentifier name="traces-sampler" />).

```

imatwawana reviewed (approved):
A couple small style changes, but otherwise, looks good!

relaxolotl committed:
fix up alerts
